### ISSUE TITLE
Voice activity XP contribution 

### ISSUE THREAD

**Issue by Fripe070** [randomairborne/experienced#199]
## Problem
There are multiple cases where a server member might be far more active in voice channels than in text channels. At the moment, voice activity is at all factored into the XP calculation, leading to these members having extremely low levels despite being some of the most active ones.

## Proposed solution
To solve this issue, I propose that voice activity (perhaps only during microphone activation? I can’t recall if that’s visible to the bot without being in the call) contributes to a secondary experience stat, that is added onto the currently existing XP system upon level calculation. 
The advantage of it being a separately tracked statistic is that it is trivial to dynamically change how much influence voice activity should have upon the calculated levels on a per-server configurable basis.

**Comment by Fripe070** [comment]
In case a separately tracked statistic would be too annoying to implement, it might be fine to pre-multiply the XP when adding it? Though I’m unsure if that would work too well

**Comment by randomairborne** [comment]
A separately tracked stat would be perfectly fine to implement. I think the annoying part here is just tracking voice states.

Do you want to work on this at all?

**Comment by Fripe070** [comment]
> Do you want to work on this at all?

I'm not familiar with how experienced is structured, but maybe. I wouldn't count on it though

**Comment by randomairborne** [comment]
I can help if you wish. I might also be able to work on this at some point.

**Comment by strich3** [comment]
Is this still planned?

**Comment by randomairborne** [comment]
Yes, this is still planned for someday. I have a few other things I want to get to first, and the demand for this has seemed pretty low. If you really want it, I will get to it- however, if you know a little rust, I can also absolutely coach you through working on the codebase.

**Comment by randomairborne** [comment]
Thinking about the design for this- my main concern is giving people new roles? that seems like it would be a lot of database requests made passively? not sure

**Comment by randomairborne** [comment]
On the other hand, database load is currently very low. So it may not be a problem. The other problem is just tracking who's in voice, and having an interval timer for each- not sure how i feel about running an arbitrary number of background tasks? on the other hand, that's what tokio is designed for.
